tests/test_execution.py

```diff
@@ -315,6 +320,22 @@ def test_aggregate_composite_traintuples(global_execution_env):
         )
         sessions[0].add_testtuple(spec).future().wait()
 
+    if not network.options.enable_intermediate_model_removal:
+        return
+
+    # Optional (if "enable_intermediate_model_removal" is True): ensure the aggregatetuple of round 1 has been deleted
+    session = sessions[0]
+    dataset = session.state.datasets[0]
+    algo = session.add_algo(spec)
+
+    spec = factory.create_traintuple(
+        algo=algo,
+        dataset=dataset,
+        data_samples=dataset.train_data_sample_keys,
+    )
+    traintuple = session.add_traintuple(spec).future().wait()
+    assert traintuple.status == assets.Status.failed
```

Comment:
Ideally it would be better to try to do a request "as a backend" to get the deleted model.
This will be closer to what we want to test and would also check that this request is correctly handled when the model has been deleted. Here, we cannot know for sure the failure reason.

Unfortunately this cannot be done now as the username/password are not available in the settings files.

Reply:
Good point, I'll add a comment to reflect this!